website/pages/api-docs/system/internal-counters.mdx

```diff
@@ -130,3 +130,166 @@ $ curl \
+
+All namespaces will be included, that are children of the namespace where the request was made.
+
+If some namespace has subsequently been deleted, its path will be listed as "deleted namespace :ID:".
+
+This endpoint was added in Vault 1.6.
+
+| Method | Path                              |
+| :----- | :-------------------------------- |
+| `GET`  | `/sys/internal/counters/activity` |
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    http://127.0.0.1:8200/v1/sys/internal/counters/activity
+```
+
+### Sample Response
+
+```json
+{
+  "request_id": "26be5ab9-dcac-9237-ec12-269a8ca647d5",
+  "lease_id": "",
+  "renewable": false,
+  "lease_duration": 0,
+  "data": {
+    "start_time": "2019-11-01T00:00:00Z",
+    "end_time": "2020-10-31T23:59:59Z",
+    "total": {
+      "distinct_entities": 90,
+      "non_entity_tokens": 130,
+      "clients": 220
+    }
+    "by_namespace": [
+      {
+        "namespace_id": "root",
+        "namespace_path": "",
+        "counts": {
+          "distinct_entities": 85,
+          "non_entity_tokens": 15,
+          "clients": 100
+        }
+      },
+      {
+        "namespace_id": "DochC",
+        "namespace_path": "ns2/",
+        "counts": {
+          "distinct_entities": 0,
+          "non_entity_tokens": 100,
+          "clients": 100
+        }
+      },
+      {
+        "namespace_id": "RtgpW",
+        "namespace_path": "ns1/",
+        "counts": {
+          "distinct_entities": 5,
+          "non_entity_tokens": 15,
+          "clients": 20
+        }
+      },
+    ],
+  },
+  "wrap_info": null,
+  "warnings": null,
+  "auth": null
+}
+```
+
+### Sample request for a single month
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    http://127.0.0.1:8200/v1/sys/internal/counters/activity?end_time=2020-06-30T00%3A00%3A00Z&start_time=2020-06-01T00%3A00%3A00Z 
+```
+
+Times must be given as RFC3339 timestamps; both `start_time` and `end_time` are optional.
+
+## Update the Client Count Configuration
+
+The `/sys/internal/counters/config` endpoint is used to configure logging of active clients.
+
+| Method  | Path                              |
+| :------ | :-------------------------------- |
+| `POST`  | `/sys/internal/counters/config`   |
+
+### Parameters
+
+- `default_report_months` `(integer: 12)` - The number of months to report if no `start_time` is specified in a query.
+- `enabled` `(string: enable, disable, default)` - Enable or disable counting of client activity. When set to `default`, the client
+counts are enabled on Enterprise builds and disabled on OSS builds. Disabling the feature during the middle of a month will
+discard any data recorded for that month, but does not delete previous months.
+- `retention_months` `(integer: 24)` - The number of months of history to retain.
+
+### Sample Payload
+
+```json
+{
+  "enabled" : "enable",
+  "default_report_months" : 12
```

Comment:
is it valuable to also include `retention_months` in the example?

Reply:
I wanted an example that showed not all the parameters were required, but that might be too subtle.  :)

Reply:
you could call out optional parameters above in their description?

this got me thinking, what's the behavior on primary if not all parameters are provided? will it set the non-provided ones to their default values, or will it retain what they already were? And further, which behavior do we want?

Reply:
The typical Vault convention is that partial updates are allowed; unfortunately there may be exceptions, but generally a POST to a configuration is a "patch".